Framework/Core/src/LifetimeHelpers.cxx

```diff
@@ -117,24 +117,34 @@ ExpirationHandler::Creator LifetimeHelpers::timeDrivenCreation(std::chrono::micr
     if (*last == 0ULL || (index.didReceiveData() == false && timerHasFired)) {
       std::random_device r;
       std::default_random_engine e1(r());
-      std::uniform_int_distribution<uint64_t> dist(0, period.count() * 0.9);
-      auto randomizedPeriodUs = static_cast<int64_t>(dist(e1) + period.count() * 0.1);
+      std::uniform_int_distribution<uint64_t> dist(0, periods.front().count() * 0.9);
+      auto randomizedPeriodUs = static_cast<int64_t>(dist(e1) + periods.front().count() * 0.1);
       *last = getCurrentTime() - randomizedPeriodUs;
       updateTimerPeriod(randomizedPeriodUs / 1000, randomizedPeriodUs / 1000);
       *stablePeriods = false;
       LOG(debug) << "Timer updated to a randomized period of " << randomizedPeriodUs << "us";
     } else if (timerHasFired && *stablePeriods == false) {
-      updateTimerPeriod(period.count() / 1000, period.count() / 1000);
+      updateTimerPeriod(periods.front().count() / 1000, periods.front().count() / 1000);
       *stablePeriods = true;
-      LOG(debug) << "Timer updated to a stable period of " << period.count() << "us";
+      LOG(debug) << "Timer updated to a stable period of " << periods.front().count() << "us";
     }
     // Nothing to do if the time has not expired yet.
     if (timerHasFired == false) {
       auto newOldest = index.setOldestPossibleInput({*last}, channelIndex);
       index.updateOldestPossibleOutput();
       return TimesliceSlot{TimesliceSlot::INVALID};
     }
+    // Get the first time we were invoked.
+    static auto firstTime = getCurrentTime();
```

Comment:
will that work with multithreading data processors in one process?

Reply:
We want something which is the first time any timer gets executed, no? If yes, than this is the correct behaviour. If not, indeed we would need to keep track about that information at the input level.

Reply:
Yes, indeed. Once `intervals` are actually `validities`, it makes sense to me, thanks.